core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -205,18 +205,18 @@ public void commit() throws IOException {
             try {
                 Files.move(tmpPath, destPath, StandardCopyOption.REPLACE_EXISTING);
             } catch (IOException e1) {
-                e1.addSuppressed(e);
+                e.addSuppressed(e1);
```

Comment:
this should not be done in the case AtomicMove is unsupported (`e instanceof AtomicMoveNotSupportedException`)- otherwise the actual failure will be suppressed rather than shown (not all file systems support atomicMove and as such all attempts will fail with that for some instances)

Reply:
You mean the fact that atomic move is not supported by the OS/FS is not the problem here, but the fact it can not be moved "normally" is. I tend to agree.

Should we only reattempt the "normal" move in case of this exception, btw?

Reply:
I don't think we should limit the attempt to only a normal move in the case of an `AtomicMoveNotSupportedException`. That may cause some previously succeeding cases to fail.

Reply:
AtomicMoveNotSupported is an expected  condition.  basically there is no API to see if you can do one before you try and do it.